src/cmap/auth/mongodb_oidc/automated_callback_workflow.ts

```diff
@@ -0,0 +1,61 @@
+import { type Document } from 'bson';
+
+import { MongoMissingCredentialsError } from '../../../error';
+import { type Connection } from '../../connection';
+import { type AuthMechanismProperties, type MongoCredentials } from '../mongo_credentials';
+import { type OIDCCallbackFunction } from '../mongodb_oidc';
+import { CallbackWorkflow } from './callback_workflow';
+import { type TokenCache, type TokenEntry } from './token_cache';
+
+const NO_CALLBACK = 'No OIDC_CALLBACK provided for callback workflow.';
+
+/**
+ * Class implementing behaviour for the non human callback workflow.
+ * @internal
+ */
+export class AutomatedCallbackWorkflow extends CallbackWorkflow {
+  /**
+   * Execute the OIDC callback workflow.
+   */
+  async execute(
+    connection: Connection,
+    credentials: MongoCredentials,
+    cache?: TokenCache
+  ): Promise<Document> {
```

Comment:
This has been refactored and now result is used.

Reply:
Ah I missed that. I went now and refactored the `Workflow` interface to make `execute` and `reauthenticate` return `Promise<void>` and then made the changes down the object model, since they don't need to return a document at all.